template/library_test.go

```diff
@@ -68,16 +69,59 @@ func TestTemplateV1Registration(t *testing.T) {
 }
 
 func TestTemplateV2Registration(t *testing.T) {
-	if _, err := GetV2("foo", false); err == nil {
+	if _, err := GetV2("foo", false, ""); err == nil {
 		t.Error("expected Get to return an error for a non-existant template")
 	}
 	if err := addV2(template{name: "foo", composeYmlTmpl: "someyaml"}); err != nil {
 		t.Error("expected Add to register a template")
 	}
-	if _, err := GetV2("foo", false); err != nil {
+	if _, err := GetV2("foo", false, ""); err != nil {
 		t.Error("expected Get to return yaml after registering a template")
 	}
 	if err := addV2(template{name: "foo", composeYmlTmpl: "someyaml"}); err == nil {
 		t.Error("expected Add to return an error when registering an existing template")
 	}
 }
+
+func TestOverridingTemplateImage(t *testing.T) {
+  if err := addV1(template{name: "baz", composeYmlTmpl: "someyaml"}); err != nil {
+    t.Error("expected Add to register a template")
+  }
+
+  if tmplyaml, err := GetV1("baz", false, ""); err != nil {
+    t.Error("expected Get to return yaml after registering a template")
+  } else {
+    if (strings.Contains(tmplyaml, "image: bazimage")) {
+      t.Error("template should not have had image in it")
+    }
+  }
+
+  if tmplyaml, err := GetV1("baz", false, "bazimage"); err != nil {
+    t.Error("expected Get to return yaml after registering a template")
+  } else {
+    if (strings.Contains(tmplyaml, "image: bazimage")) {
```

Comment:
How does this test pass, doesn't the template_image interpolation logic rely on the `template.composeYmlTmpl` having `{{.TemplateImage}}` inside it?